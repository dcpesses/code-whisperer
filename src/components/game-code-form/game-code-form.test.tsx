import {render, screen, fireEvent} from '@testing-library/react';
import GameCodeForm from './index';

describe('GameCodeForm', () => {
  test('Should render as expected', () => {
    const {container} = render(<GameCodeForm />);

    expect(container).toMatchSnapshot();
  });

  test('Should render and handle focus/input change/click events when enabled', () => {

    const onInputChange = vi.fn();
    const onSendToAll = vi.fn();

    render(<GameCodeForm disabled={false} onInputChange={onInputChange} onSendToAll={onSendToAll} />);

    const mockFocusEvent = {target: {select: vi.fn()}};
    const mockChangeEvent = {target: {value: 'E'}};

    const inputRoomCode = screen.getByPlaceholderText('ENTER ROOM CODE');
    const btnSendToQueue = screen.getByRole('button');

    fireEvent.focus(inputRoomCode, mockFocusEvent);
    fireEvent.change(inputRoomCode, mockChangeEvent);
    fireEvent.click(btnSendToQueue);

    expect(mockFocusEvent.target.select).toHaveBeenCalledTimes(1);
    expect(onInputChange).toHaveBeenCalledTimes(1);
    expect(onSendToAll).toHaveBeenCalledTimes(1);
  });

  test('should not handle click events when disabled', () => {
    const onSendToAll = vi.fn();

    const {rerender} = render(<GameCodeForm disabled={false} onSendToAll={onSendToAll} />);

    const mockFocusEvent = {target: {select: vi.fn()}};
    const mockChangeEvent = {target: {value: 'E'}};
    const inputRoomCode = screen.getByPlaceholderText('ENTER ROOM CODE');
    const btnSendToQueue = screen.getByRole('button');

    fireEvent.focus(inputRoomCode, mockFocusEvent);
    fireEvent.change(inputRoomCode, mockChangeEvent);
    fireEvent.click(btnSendToQueue);

    rerender(<GameCodeForm disabled={false} onSendToAll={onSendToAll} />);

    expect(onSendToAll).toHaveBeenCalledTimes(1);
  });
  test('should handle events using defaults', () => {
    render(<GameCodeForm disabled={false} />);

    const mockFocusEvent = {target: {select: vi.fn()}};
    const mockChangeEvent = {target: {value: 'E'}};
    const inputRoomCode = screen.getByPlaceholderText('ENTER ROOM CODE');
    const btnSendToQueue = screen.getByRole('button');

    fireEvent.focus(inputRoomCode, mockFocusEvent);
    fireEvent.change(inputRoomCode, mockChangeEvent);
    fireEvent.click(btnSendToQueue);
  });
});
