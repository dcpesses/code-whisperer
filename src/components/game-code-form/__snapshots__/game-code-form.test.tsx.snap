// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GameCodeForm > Should render as expected 1`] = `
<div>
  <div
    class="d-flex align-items-center p-3 my-3 text-white bg-purple rounded shadow-sm libre-franklin-font"
  >
    <div
      class="input-group"
    >
      <input
        aria-describedby="btn-send-to-queue"
        aria-label="Enter Code"
        class="form-control libre-franklin-font"
        placeholder="ENTER CODE"
        type="text"
      />
      <button
        class="btn btn-primary libre-franklin-font text-uppercase"
        id="btn-send-to-queue"
        type="button"
      >
        Send to Queue
      </button>
    </div>
  </div>
</div>
`;
